agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/LiveMetricsSpanProcessor.java

```diff
@@ -21,95 +21,41 @@
-      @Nullable String roleName,
-      @Nullable String roleInstance,
-      String sdkVersion) {
-    this.instrumentationKey = instrumentationKey;
-    quickPulse =
-        QuickPulse.create(
-            httpPipeline,
-            () -> endpointUrl,
-            () -> instrumentationKey,
-            roleName,
-            roleInstance,
-            true,
-            sdkVersion);
+  public LiveMetricsSpanProcessor(QuickPulse quickPulse, SpanDataMapper mapper) {
+    this.quickPulse = quickPulse;
+    this.mapper = mapper;
   }
 
   @Override
-  public void onStart(Context context, ReadWriteSpan readWriteSpan) {
-    SpanData spanData = readWriteSpan.toSpanData();
-    SpanKind kind = spanData.getKind();
-    String instrumentationName = spanData.getInstrumentationLibraryInfo().getName();
-
-    List<TelemetryItem> telemetryItems = new ArrayList<>();
-    if (kind == SpanKind.INTERNAL) {
-      if (instrumentationName.startsWith("io.opentelemetry.spring-scheduling-")
-          && !spanData.getParentSpanContext().isValid()) {
-        // if (!span.getParentSpanContext().isValid()) {
-        // TODO (trask) AI mapping: need semantic convention for determining whether to map INTERNAL
-        // to request or dependency (or need clarification to use SERVER for this)
-        TelemetryMappingHelper.addRequestTelemetryItem(
-            spanData, instrumentationKey, telemetryItems);
-      } else {
-        TelemetryMappingHelper.addRemoteDependencyTelemetryItem(
-            spanData, true, instrumentationKey, telemetryItems);
-      }
-    } else if (kind == SpanKind.CLIENT || kind == SpanKind.PRODUCER) {
-      TelemetryMappingHelper.addRemoteDependencyTelemetryItem(
-          spanData, false, instrumentationKey, telemetryItems);
-    } else if (kind == SpanKind.CONSUMER
-        && "receive".equals(spanData.getAttributes().get(SemanticAttributes.MESSAGING_OPERATION))) {
-      TelemetryMappingHelper.addRemoteDependencyTelemetryItem(
-          spanData, false, instrumentationKey, telemetryItems);
-    } else if (kind == SpanKind.SERVER || kind == SpanKind.CONSUMER) {
-      TelemetryMappingHelper.addRequestTelemetryItem(spanData, instrumentationKey, telemetryItems);
-    } else {
-      throw LOGGER.logExceptionAsError(new UnsupportedOperationException(kind.name()));
-    }
-
-    for (TelemetryItem telemetryItem : telemetryItems) {
-      quickPulse.add(telemetryItem);
-    }
-  }
+  public void onStart(Context context, ReadWriteSpan readWriteSpan) {}
 
   @Override
   public boolean isStartRequired() {
     return false;
   }
 
   @Override
-  public void onEnd(ReadableSpan readableSpan) {}
+  public void onEnd(ReadableSpan readableSpan) {
+    if (quickPulse.isEnabled()) {
+      // TODO (trask) can we do anything better here in terms of double conversion?
+      quickPulse.add(mapper.map(readableSpan.toSpanData()));
+    }
+  }
 
   @Override
   public boolean isEndRequired() {
-    return false;
+    return true;
```

Comment:
is this intended to return true now? i'm not familiar with isEndRequired.

Reply:
ya, the OTel SDK calls `isEndRequired` and only calls `onEnd` if it returns true